### PR TITLE
Allow items to have achievement, collectible and statistic requirements

### PR DESCRIPTION
**Pull request by Kookiecraft**
Syntax:
`aReqGambler=5` in txt files or `"aReqGambler": 5`  in json files (Gambler tier 5 achievement required)
`cReqBasilisk_Scale=2` in txt files or `"cReqBasilisk_Scale": 2` in json files (Basilisk Scale x2 required)
`sReqRegions*Astral_Town=1000` in txt files or `"sReqRegions*Astral_Town": 1000` in json files (1000 steps in Astral Town region required)

In short, all spaces are replaced with underscores, capitalization matters, and statistic levels are separated by *. More examples below:

`sReqBossKills*King_Minotaur=3` or `sReqBoss_Kills*King_Minotaur=3 `(either works)
`sReqCharacter*Choice*Choose*Double_Down=100`
`sReqCombat*Utilize*Digital=50000`
`sReqMaps*Norkos=1000000`
`cReqKey_to_Maeles=1`
`aReqProfessional:_Jester=5`

If an item somehow ends up with `aReqGambler=5` and `aReqGambler=6`, the final outcome is the higher one (requires Gambler tier 6). This applies for all types of requirements.

If an item has a requirement that the player does not meet, the item is not offered as a choice to the player at all (may or may not change in the future).

P.S. A new idea just dawned on me, but I haven't thought about how to implement it. Perhaps there should be "OR" conditions. i.e. make an item require either 2 gold bank coins or 2 evil bank coins.

### CODE REVIEW

File: src/core/base/character.js [hunk 1]
```diff
@@ -172,6 +172,7 @@ export class Character {
   }
 
   canEquip(item, rangeBoostMultiplier = 1, useCheckRangeMultiplier = true) {
+    const player = this;
```
Comment (seiyria) [inline src/core/base/character.js]: is this aliased for any particular reason? arrow functions auto-alias `this` to the outer scope so this is redundant in most cases.

**Comment by seiyria** [comment]
this looks good. very high quality!

File: src/core/base/character.js (hunk 1 above)
Comment (Kookiecraft) [inline src/core/base/character.js]: I needed it for the code I had previously. It was extremely dirty, and needed that reference. I'll remove it on the next PR if its still there.

Comment (seiyria) [inline src/core/base/character.js]: Sounds good to me.

**Comment by JukeDragh** [comment]
Maybe have a collective total version too. Something like:
sReqRegion*Maeles_Town+Maelian_Merchant's_Hall=1000

**Comment by seiyria** [comment]
And and or is not officially specced out at this time. I'm not sure if it's
necessary atm. We'll see if the need arises later though.

**Pull request merged**